ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,1094 @@
+                "\n        if(Identity.scenario == \"All\") return filteredDataCube.Select(x => x.Scenario == null ? x with {Scenario = \"Best Estimate\" } : x).ToDataCube();",
+                "\n        var bestEstimateById = filteredDataCube.Where(x => x.Scenario == null).ToDictionary(x => x.ToIdentityString());",
+                "\n        return filteredDataCube.Select(x => x.Scenario == null ? x with { Scenario = \"Best Estimate\" } ",
+                "\n                                                : x with { Value = x.Value - (bestEstimateById.TryGetValue((x with {Scenario = null}).ToIdentityString(), out var be)? be.Value : 0.0) }).ToDataCube();",
+                "\n    }}",
+                "\n} "
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "[InitializeScope(nameof(Init))]",
+                "\npublic interface ReportScope : IMutableScope<string>, ReportingNodeFormsEntity, MonthlyPeriodFormsEntity, ScenarioFormsEntity, CurrencyFormsEntity, ",
+                "\n                               BasicSliceAndDiceRowsFormsEntity, BasicSliceAndDiceColumnsFormsEntity, BasicFilterFormsEntity {",
+                "\n    protected IPivotFactory report => GetStorage().Report;",
+                "\n    protected IExportVariable export => GetStorage().Export;",
+                "\n    protected int headerColumnWidthValue => 250;",
+                "\n",
+                "\n    HashSet<(ReportIdentity, CurrencyType)> GetDataIdentities() => GetStorage().GetIdentities((Year, Month), ReportingNode, Scenario, CurrencyType); // TODO, add filter for identities, if the property is exposed at this level",
+                "\n",
+                "\n    IDataCube<ReportVariable> GetData() => default;",
+                "\n",
+                "\n    async Task<GridOptions> ToReportAsync() {",
+                "\n        await GetStorage().InitializeAsync((Year, Month), ReportingNode, Scenario, CurrencyType);",
+                "\n        var dataScope = GetScope<Data>(((Year, Month), ReportingNode, Scenario, CurrencyType, dataFilter));",
+                "\n        var dataCube = GetData();",
+                "\n        // Find another way to avoid nasty errors if the DC is empty - AK",
+                "\n        if (dataCube.ToArray().Count() == 0) return null;",
+                "\n        dataScope.InputDataCube = dataCube;",
+                "\n        return await GetReportTaskAsync(dataScope.DataCube);",
+                "\n    }",
+                "\n",
+                "\n    // Using this routine is highly discouraged due to the mutlithreading issue -A.K.",
+                "\n    // Avoid using these methods if working with the DB -- it will trigger synchronization error in access to the DB",
```

Comment:
This issue and comment must be addressed before the PR, imo

Reply:
Daniel, I am not sure that it can be addressed before the PR. It is a known issue, which is part of a bigger multi threading issue. The platform people are working on it, bit it will not be ready this week.

Reply:
Unfortunately this issues will not be solved before this PR. We are going to release this version, knowing that this behavior will be fixed.